src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -926,6 +926,29 @@ def list_traffic_manager_profiles(resource_group_name=None):
     else:
         return ncf.list_all()
 
+def update_traffic_manager_profile(instance, profile_status=None, routing_method=None, tags=None,
+                                   monitor_protocol=None, monitor_port=None, monitor_path=None,
+                                   ttl=None):
+    if tags is not None:
+        instance.tags = tags
+    if profile_status is not None:
+        instance.profile_status = profile_status # TODO: Needs choice list Enabled/Disabled
+    if routing_method is not None:
+        instance.traffic_routing_method = routing_method
+    if ttl is not None:
+        instance.dns_config.ttl = ttl
+
+    monitor_params = [monitor_protocol, monitor_port, monitor_path]
+    if any([x for x in monitor_params if x is not None]):
```

Comment:
Suggest get rid of this `if`, this list comprehension might cost more cycles than the 3 individual if conditions before